### PR TITLE
[3.x] Add error messages `String::hex_to_int` functions, and accept capital X in prefix

### PR DESCRIPTION
**Pull request by Rubonnek**
3.x equivalent of #54078

*Bugsquad edit:* Related to #54083.

### CODE REVIEW

**Review by AaronRecord — CHANGES_REQUESTED** [review]

File: core/ustring.cpp [hunk 1]
```diff
@@ -1773,13 +1766,13 @@ int64_t String::hex_to_int64(bool p_with_prefix) const {
 		} else if (c >= 'a' && c <= 'f') {
 			n = (c - 'a') + 10;
 		} else {
-			return 0;
+			ERR_FAIL_COND_V_MSG(true, 0, "Invalid hexadecimal notation character \"" + chr(*s) + "\" in string \"" + *this + "\".");
```
Comment (akien-mga) [inline core/ustring.cpp:1769]:
```suggestion
			ERR_FAIL_V_MSG(0, "Invalid hexadecimal notation character \"" + chr(*s) + "\" in string \"" + *this + "\".");
```

File: core/ustring.cpp [hunk 2]
```diff
@@ -1730,36 +1726,33 @@ int String::hex_to_int(bool p_with_prefix) const {
...
```
Comment (akien-mga) [inline core/ustring.cpp:1729]:
```suggestion
			ERR_FAIL_V_MSG(0, "Invalid hexadecimal notation character \"" + chr(*s) + "\" in string \"" + *this + "\".");
```

File: core/ustring.cpp [hunk 3]
```diff
@@ -1701,22 +1701,18 @@ String::String(const StrRange &p_range) {
 
 int String::hex_to_int(bool p_with_prefix) const {
 	int len = length();
-	if (len == 0 || (p_with_prefix && len < 3)) {
-		return 0;
-	}
+	ERR_FAIL_COND_V_MSG(p_with_prefix ? len < 3 : len == 0, 0, String("Invalid hexadecimal notation length in string ") + (p_with_prefix ? "with" : "without") + "prefix \"" + *this + "\".");
```
Comment (akien-mga) [inline core/ustring.cpp:1704]:
```suggestion
	ERR_FAIL_COND_V_MSG(p_with_prefix ? len < 3 : len == 0, 0, String("Invalid hexadecimal notation length in string ") + (p_with_prefix ? "with" : "without") + " prefix \"" + *this + "\".");
```

File: core/ustring.cpp [hunk 4]
```diff
@@ -1730,36 +1726,33 @@ int String::hex_to_int(bool p_with_prefix) const {
...
 
 	return hex * sign;
 }
 
 int64_t String::hex_to_int64(bool p_with_prefix) const {
-	if (p_with_prefix && length() < 3) {
-		return 0;
-	}
+	int len = length();
+	ERR_FAIL_COND_V_MSG(p_with_prefix ? len < 3 : len == 0, 0, String("Invalid hexadecimal notation length in string ") + (p_with_prefix ? "with" : "without") + "prefix \"" + *this + "\".");
```
Comment (akien-mga) [inline core/ustring.cpp:1744]:
```suggestion
	ERR_FAIL_COND_V_MSG(p_with_prefix ? len < 3 : len == 0, 0, String("Invalid hexadecimal notation length in string ") + (p_with_prefix ? "with" : "without") + " prefix \"" + *this + "\".");
```

**Review by akien-mga — APPROVED** [review]
Looks great!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**